### PR TITLE
[doxygen] Add versions 1.9.7 and 1.9.8.

### PR DESCRIPTION
**Pull request by KineticTheory**
* Adding 2 new versions.
* `1.9.6` has a [known bug](https://github.com/doxygen/doxygen/issues/10165) that impacts my projects that is fixed in these two newer releases.

### CODE REVIEW

File: var/spack/repos/builtin/packages/doxygen/package.py [hunk 1]
```diff
@@ -18,6 +18,8 @@ class Doxygen(CMakePackage):
     homepage = "https://www.doxygen.org"
     url = "https://github.com/doxygen/doxygen/archive/refs/tags/Release_1_9_5.tar.gz"
 
+    version("1.9.8", sha256="691777992a7240ed1f822a5c2ff2c4273b57c1cf9fc143553d87f91a0c5970ee")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/doxygen/package.py]: How did you calculate this?  

I'm getting the following from "spack checksum doxygen 1.9.8" instead:
`77371e8a58d22d5e03c52729844d1043e9cbf8d0005ec5112ffa4c8f509ddde8`

**Review by tldahlgren — COMMENTED** [review]
Confirmed only one of the two version sha256.

File: var/spack/repos/builtin/packages/doxygen/package.py (hunk 1 above)
Comment (KineticTheory) [reply var/spack/repos/builtin/packages/doxygen/package.py]: Yes.  Stupid cut&paste error.  I have your has in my editor but copied the wrong one.  It should be fixed in [f7bd57c](https://github.com/spack/spack/pull/41123/commits/f7bd57cdb9f54eacd26e1e4aad309154ba01c6a5).

**Review by tldahlgren — APPROVED** [review]
Confirmed both sha256.

**Pull request merged**